### PR TITLE
Feature/41 events

### LINKED ISSUE #41: Operator shows dogu install/delete/upgrade states as events

### PR DESCRIPTION
**Pull request by Subaro**
Resolves #41

### CODE REVIEW

File: docs/development/event_policy_for_the_operator_de.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Event-Richtlinie für den `k8s-dogu-operator`
+
+## Was ist ein Kubernetes-Event?
+
+Ein Event ist eine Möglichkeit, den Benutzer über Vorgänge in Bezug auf andere Clusterobjekte zu informieren. Ein
+perfektes Beispiel für ein Event ist die Installation eines Dogu im Cluster. Im Allgemeinen werden die
```
Comment (nhinze23) [inline docs/development/event_policy_for_the_operator_de.md:6]: Dogu's

File: docs/development/event_policy_for_the_operator_de.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+verantwortliche Person zu unterstützen.
+
+Eine große Stärke bei der Verwendung von Events ist ihre Flexibilität. Es ist recht einfach, die Events
+bezüglich eines Dogus aus dem Cluster zu lesen und seinen aktuellen Zustand zu verstehen. Es ist auch möglich, auf
+bestimmte Events zu reagieren, indem beliebige Aufgaben ausgeführt werden.
+
+## Umfang der Events im `k8s-dogu-operator`
+
+Der dogu-Operator erstellt Events für die dogu-CRD, während er tägliche Aufgaben wie die Installation, Löschung oder
+Aktualisierung eines dogu durchführt. Die meisten Prozessinformationen in der dogu-Ressource helfen dabei, den aktuellen
```
Comment (nhinze23) [inline docs/development/event_policy_for_the_operator_de.md:19]: Dogu's

Comment (nhinze23) [inline docs/development/event_policy_for_the_operator_de.md:19]: Dogu-Ressource

File: docs/development/event_policy_for_the_operator_de.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+![Bild, das die Ereignisse beim Auftreten eines Fehlers bei der Installation des `postgresql` Dogu zeigt.](figures/events_with_errors.png)
+
+**Erfolgreiche `postgresql`-Dogu-Installation**
+
+![Bild mit Ereignissen für die erfolgreiche Installation des `postgresql` Dogu.](figures/events_without_errors.png)
+
+## Verwendung von Ereignissen im `k8s-dogu-operator`
+
+Die [kubebuilder Dokumentation](https://book-v1.book.kubebuilder.io/beyond_basics/creating_events.html) erklärt
+perfekt, wie man Events innerhalb eines Kubernetes-Controllers verwendet.
```
Comment (nhinze23) [inline docs/development/event_policy_for_the_operator_de.md:38]: -perfekt

File: docs/development/event_policy_for_the_operator_de.md (hunk 1 above)
Comment (nhinze23) [inline docs/development/event_policy_for_the_operator_de.md:6]: -perfekt

File: main.go [hunk 4]
```diff
@@ -57,9 +57,35 @@ var (
 func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 	utilruntime.Must(k8sv1.AddToScheme(scheme))
+	utilruntime.Must(AddFieldLabelConversionsForEvent(scheme))
+
 	//+kubebuilder:scaffold:scheme
 }
```
Comment (nhinze23) [inline main.go:64]: missing comment

File: controllers/dogu_controller.go [hunk 5]
```diff
@@ -113,18 +128,22 @@ func (r *doguReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
...
-			}
+		})
+		if err != nil {
+			r.recorder.Event(doguResource, v1.EventTypeWarning, ErrorOnRequeueEventReason, "Failed to requeue the installation.")
+			return ctrl.Result{}, fmt.Errorf("failed to handle requeue: %w", err)
 		}
+
 		return result, nil
 	case Upgrade:
 		return ctrl.Result{}, nil
```
Comment (nhinze23) [inline controllers/dogu_controller.go:149]: Use the event recorder in the delete case as well.

File: controllers/dogu_manager.go [hunk 6]
```diff
@@ -162,15 +165,18 @@ func deleteDoguConfigMap(ctx context.Context, client client.Client, doguConfigMa
 
 // Install installs a dogu resource.
 func (m *DoguManager) Install(ctx context.Context, doguResource *k8sv1.Dogu) error {
+	m.recorder.Event(doguResource, "Normal", "Installation", "Starting installation...")
 	return m.installManager.Install(ctx, doguResource)
 }
 
 // Upgrade upgrades a dogu resource.
 func (m *DoguManager) Upgrade(_ context.Context, _ *k8sv1.Dogu) error {
+	//m.recorder.Eventf(doguResource, "Normal", "Installation", "Starting upgrade of dogu [%s] from version [%s] to version [%s]", doguResource.Name)
```
[the remark this answers is not in the document]
Comment (Subaro) [reply controllers/dogu_manager.go:174]: Done

**Comment by nhinze23** [comment]
lgtm

**Pull request merged**